sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -688,6 +688,36 @@ abstract class SparkPlanMeta[INPUT <: SparkPlan](plan: INPUT,
     // shuffled exchanges. So broadcast exchanges are not impacted which could have an impact on
     // BroadcastHashJoin, and shuffled exchanges are not used to disable anything downstream.
     fixUpExchangeOverhead()
+
+    // 3) Some child nodes can't run on GPU if parent nodes can't run on GPU.
+    // WriteFilesExec is a new operator from Spark version 340,
+    // Did not extract a shim code for simplicity
+    tagChildAccordingToParent(this.asInstanceOf[SparkPlanMeta[SparkPlan]], "WriteFilesExec")
+  }
+
+  /**
+   * tag child node can't run on GPU if parent node can't run on GPU and child node is a `typeName`
+   * From Spark 340, plan is like:
+   *    InsertIntoHadoopFsRelationCommand
+   *    +- WriteFiles
+   *      +- sub plan
+   * Instead of:
+   *    InsertIntoHadoopFsRelationCommand
+   *    +- sub plan
+   * WriteFiles is a temporary node and does not have input and output, it acts like a tag node.
+   * @param p        plan
+   * @param typeName type name
+   */
+  def tagChildAccordingToParent(p: SparkPlanMeta[SparkPlan], typeName: String): Unit = {
+    p.childPlans.foreach(e => tagChildAccordingToParent(e, typeName))
+    if (p.wrapped.getClass.getSimpleName.equals(typeName)) {
+      assert(p.parent.isDefined)
+      if (!p.parent.get.canThisBeReplaced) {
+        // parent can't run on GPU, also tag this.
+        p.willNotWorkOnGpu(
+          "WriteFilesExec can't run on GPU because parent can't run on GPU")
```

Comment:
Seems it is not good to hard code the `WriteFilesExec` in the message string.
```suggestion
          s"$typeName can't run on GPU because its parent can't run on GPU")
```